cfme/tests/cloud_infra_common/test_provisioning.py

```diff
@@ -378,38 +390,57 @@ def test_cloud_provision_from_template_with_attached_disks(
     if provider.one_of(AzureProvider):
         recursive_update(inst_args, {'environment': {'availability_zone': provisioning("av_set")}})
 
-    device_name = "/dev/sd{}"
+    device_name = "vd{}"
     device_mapping = []
 
-    with provider.mgmt.with_volumes(1, n=disks) as volumes:
-        for i, volume in enumerate(volumes):
-            device_mapping.append((volume, device_name.format(chr(ord("b") + i))))
-        # Set up automate
+    volumes = provider.mgmt.volume_configurations(1, n=disks)
+
+    @request.addfinalizer
+    def delete_volumes():
+        for volume in volumes:
+            provider.mgmt.delete_volume(volume)
 
-        method = modified_request_class.methods.instantiate(name="openstack_PreProvision")
+    # Set up automate
+    for i, volume in enumerate(volumes, 0):
+        if i == 0:
+            provider.mgmt.capi.volumes.set_bootable(volume, True)
+            device_mapping.append(
+                {'boot_index': 0,
+                'uuid': volume,
+                'device_name': device_name.format(chr(ord("a") + i))})
+        else:
+            device_mapping.append(
+                {'boot_index': -1,
+                'uuid': volume,
+                'device_name': device_name.format(chr(ord("a") + i))})
 
+    method = modified_request_class.methods.instantiate(name="openstack_PreProvision")
+    with update(method):
+        disk_mapping = []
+        for mapping in device_mapping:
+            disk_mapping.append(ONE_FIELD.format(**mapping))
+        method.script = VOLUME_METHOD.format(",\n".join(disk_mapping))
+
+    @request.addfinalizer
+    def _finish_method():
         with update(method):
-            disk_mapping = []
-            for mapping in device_mapping:
-                disk_mapping.append(ONE_FIELD.format(*mapping))
-            method.script = VOLUME_METHOD.format(", ".join(disk_mapping))
-
-        def _finish_method():
-            with update(method):
-                method.script = """prov = $evm.root["miq_provision"]"""
-        request.addfinalizer(_finish_method)
-
-        instance = appliance.collections.cloud_instances.create(vm_name,
-                                                                provider,
-                                                                form_values=inst_args)
-
-        for volume_id in volumes:
-            soft_assert(vm_name in provider.mgmt.volume_attachments(volume_id))
-        for volume, device in device_mapping:
-            soft_assert(provider.mgmt.volume_attachments(volume)[vm_name] == device)
-        instance.mgmt.delete()  # To make it possible to delete the volume
+            method.script = """prov = $evm.root["miq_provision"]"""
```

Comment:
Question: reason for triple quotes here?

Reply:
again, the topic of this PR is not to beatifulize the the code. I wanted to make things working. So I was not mangling with everything I saw and could mangle.

Perhaps someone found this more readable then mixing apostrophes and quotes.

Reply:
I removed the hard encoded sttrings used for reseting the `method.script`. I changed the code so that it first stores the `method.script`, then modify, then restores it back